File: Java/Gate Four/TestLevelSix.java

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;


public class TestLevelSix{




    @Test
    public void testThatArrayIsOfSpiltIntoEvenAndOddArrays(){

    LevelSix check = new LevelSix();
    
    int [] numbers = {45,60,3,10,9,22};

    int [] actualResult = check.SortedOddEvenArrays(numbers);

    int [] result = {{45,3,9},{60,10,22}};


    assertEquals(actualResult, result);
     

    }




    @Test
    public void testThatAllPerfectSquareNumbersInTheArrayArePrinted(){

    LevelSix check = new LevelSix();
    
    int []  numbers = {45, 7, 9, 10, 16, 18, 4};

    int [] actualResult = check.PerfectSquareNumbers(numbers);

    int [] result = {4,9,16};

    assertArrayEquals(actualResult, result);
     

    }





    @Test
    public void testThatArrayIsPalindromic(){

    LevelSix check = new LevelSix();
    
    int [] numbers = {4,7,0,10,16,18};

    int [] actualResult = check.palindromicArrays(numbers);

    int [] result = {4,9,16};


    assertEquals(actualResult, result);
     

    }




    @Test
    public void testThatAllNonPerfectNumbersInTheArrayAreReplaceWithMInusONe(){

    LevelSix check = new LevelSix();
    
    int []  numbers = {4, 7, 8, 9, 10, 49, 6};

    int [] actualResult = check.NonPerfectSquareNumbersInArray(numbers);

    int [] result = {4,-1, 9, -1, 49, -1};

    assertArrayEquals(actualResult, result);
     

    }
}

File: Python/Gate Four/test_level_six.py
from unittest import TestCase

import level_six

class TestLevelSixTask(TestCase):

    def test_that_array_is_of_spilt_into_even__and_odd_arrays(self):

        numbers = [45,60,3,10,9,22]

        actual_result = level_six.sorted_odd_even_arrays(numbers)

        expected_result = [[45,3,9],[60,10,22]]

        self.assertEqual(actual_result, expected_result)

    def test_that_all_perfect_square_numbers_in_the_array_are_printed(self):

        numbers = [45, 7, 9, 10, 16, 18, 4]

        actual_result = level_six.perfect_square_numbers(numbers)

        expected_result = [4,9,16]

        self.assertEqual(actual_result, expected_result) 

#    def test_that_all_zeros_are_moved_to_end_of_the_array_list(self):
#
#        numbers = [5,0,3,0,2,0]
#
#        actual_result = level_five.zero_numbers(numbers)
#
#        expected_result = [5,3,2,0,0,0]
#
#        self.assertEqual(actual_result, expected_result) 
#
#    def test_that_all_common_and_duplicate_numbers__are_found_and_returned(self):
#
#        numbers = [45,60,3,0,67,2,45,3,22,0]
#
#        actual_result = level_five.return_duplicate_numbers(numbers)
#
#        expected_result = [45,3,0]
#
#        self.assertEqual(actual_result, expected_result) 
